src/mono/wasi/runtime/main.c

```diff
@@ -110,6 +111,14 @@ int main(int argc, char * argv[]) {
 		mono_print_unhandled_exception(out_exc);
 		exit(1);
 	}
-	return ret < 0 ? -ret : ret;
+	ret = ret < 0 ? -ret : ret;
+
+	char* dotnet_wasi_print_exit_code = monoeg_g_getenv ("DOTNET_WASI_PRINT_EXIT_CODE");
```

Comment:
@dicej @SingleAccretion 

I'm introducing this trick to be able to work with proper exit code in various tests we have.
maybe NativeAOT could do the same.
Feedback welcome.

Reply:
We have similar logic in Browser

Reply:
> new dotnet run option --forward-exit-code which sets it

This is _not_ the default for `run`, correct?

To be honest, it is sad to see this workaround being needed, but it is what it is.

Reply:
no, it's not a default